crates/storage/store.rs

```diff
@@ -311,6 +320,91 @@ impl Store {
         Ok(Some(account_state.nonce))
     }
 
+    /// Applies account updates based on the block's latest storage state
+    /// and returns the new state root after the updates have been applied.
+    pub async fn apply_account_updates_batch(
+        &self,
+        block_hash: BlockHash,
+        account_updates: &[AccountUpdate],
+    ) -> Result<
+        Option<(
+            H256,
+            Vec<(NodeHash, Vec<u8>)>,
+            Vec<(Vec<u8>, Vec<(NodeHash, Vec<u8>)>)>,
+        )>,
+        StoreError,
```

Comment:
This return type requires documentation, and I suggest declaring it outside the function, possibly as a struct.